src/providers/wms/qgswmsprovider.cpp

```diff
@@ -1130,11 +1130,20 @@ void QgsWmsProvider::createTileRequestsXYZ( const QgsWmtsTileMatrix *tm, const Q
       turl.replace( QLatin1String( "{q}" ), _tile2quadkey( tile.col, tile.row, z ) );
 
     turl.replace( QLatin1String( "{x}" ), QString::number( tile.col ), Qt::CaseInsensitive );
-    turl.replace( QLatin1String( "{y}" ), QString::number( tile.row ), Qt::CaseInsensitive );
+    // inverted Y axis
+    bool invertedY = turl.contains( QLatin1String( "{-y}" ) );
+    if ( invertedY )
+    {
+      turl.replace( QLatin1String( "{-y}" ), QString::number( tm->matrixHeight - tile.row ), Qt::CaseInsensitive );
```

Comment:
Hehe I think Nyall meant it more as a joke... anyway we can keep it as a bugfix then :-)

Feel free to merge (could you maybe just fix XYX -> XYZ in commit message?)